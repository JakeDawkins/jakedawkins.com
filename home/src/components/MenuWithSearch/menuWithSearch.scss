@import "../../styles/variables";

aside {
  &.slide {
    left: 0;
  }
  nav {
    ul {
      padding: 0;
    }
    li {
      margin: 0;
      list-style-type: none;
      a {
        width: 100%;
        display: block;
        padding: 1.25rem;
        text-decoration: none;
        font-family: $font--header;
        font-weight: $weight--normal;
        font-style: normal;
        color: $color--aside-link;
        border-top: 1px solid $color--aside-link-border;
        &:hover {
          color: $color--aside-link-hover;
        }
      }
    }
  }
}
