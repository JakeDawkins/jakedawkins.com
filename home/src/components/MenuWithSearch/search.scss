@import "../../styles/variables";

//@extend-elements
//original selectors
//.search-wrapper *zoom 1:after, .search-wrapper *zoom 1:before
%extend_1 {
  display: table;
  line-height: 0;
  content: "";
}

//original selectors
//.search-wrapper:before, .search-wrapper:after
%extend_2 {
  display: table;
  line-height: 0;
  content: "";
}

.dosearch {
  right: 10px;
  background: $color--blue;
  //Instead of the line below you could use @include border-radius($radius, $vertical-radius)
  border-radius: .188rem;
  padding: .125rem .188rem;
  position: absolute;
  display: block;
  width: 40px;
  height: 40px;
  top: 10px;
  z-index: 31;
}
.icon-search {
  width: 2.188rem;
  height: 2.188rem;
}
.search-wrapper {
  //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
  transform: translateY(-400px);
  z-index: 9999;
  position: absolute;
  top: 50px;
  width: 100%;
  padding-right: 10%;
  padding-left: 10%;
  &.active {
    //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
    transform: translateY(0);
  }
  *zoom {
    1:after {
      @extend %extend_1;
    }
    2:before {
      @extend %extend_1;
    }
    3:after {
      clear: both;
    }
  }
  &:before {
    @extend %extend_2;
  }
  &:after {
    @extend %extend_2;
  }
}
.search-form {
  position: relative;
  top: 0;
  //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
  transform: translateX(-200px);
  z-index: 9999;
  width: 100%;
  filter: alpha(opacity=0);
  opacity: 0;
  //Instead of the line below you could use @include transition($transition-1, $transition-2, $transition-3, $transition-4, $transition-5, $transition-6, $transition-7, $transition-8, $transition-9, $transition-10)
  transition: all 200ms 100ms cubic-bezier(0,.6,.4,1);
  h4 {
    margin: .625rem 0;
    color: $color--white;
  }
  .search-field {
    width: 100%;
    font-family: $font--header;
    font-weight: $weight--normal;
    font-style: normal;
    font-size: 1.625rem;
    color: $color--white;
    background-color: transparent;
    border: 0;
    border-bottom: 1px solid $color--white;
    //Instead of the line below you could use @include border-radius($radius, $vertical-radius)
    border-radius: 0;
    //Instead of the line below you could use @include box-shadow($shadow-1, $shadow-2, $shadow-3, $shadow-4, $shadow-5, $shadow-6, $shadow-7, $shadow-8, $shadow-9, $shadow-10)
    box-shadow: none;
    //Instead of the line below you could use @include background-clip($clip)
    background-clip: padding-box;
    &:focus {
      outline: 0;
      //Instead of the line below you could use @include box-shadow($shadow-1, $shadow-2, $shadow-3, $shadow-4, $shadow-5, $shadow-6, $shadow-7, $shadow-8, $shadow-9, $shadow-10)
      box-shadow: none;
    }
  }
  &.active {
    top: 0;
    //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
    transform: translateX(0);
    filter: none;
    opacity: 1;
  }
  &.hidden {
    display: none;
  }
  .icon-remove-sign {
    position: absolute;
    top: 0;
    right: 0;
    z-index: 1;
    display: block;
    fill: $color--white;
    width: 30px;
    height: 30px;
    text-align: center;
    cursor: pointer;
  }
  .search-field::-webkit-search-decoration {
    display: none;
  }
  .search-field::-webkit-search-cancel-button {
    display: none;
  }
  .search-field::-webkit-search-results-button {
    display: none;
  }
  .search-field::-webkit-search-results-decoration {
    display: none;
  }
  .search-list {
    position: absolute;
    width: 100%;
    font-family: $font--body;
    font-weight: $weight--normal;
    font-style: normal;
  }
  li {
    color: $color--white;
  }
  p {
    color: $color--white;
  }
  a {
    color: $color--white;
    &:hover {
      color: $color--white;
    }
  }
}
.search-form::-webkit-input-placeholder {
  font-size: 1.625rem;
}
@media only screen and(max-width:37.5rem) {
  body.search-overlay:after {
    background-color: $color--blue;
  }
}
.search-list {
  padding: 0;
  margin: 0;
  list-style-type: none;
  .entry-date {
    float: right;
    display: none;
    font-size: 14px;
    text-transform: uppercase;
  }
  a {
    text-decoration: none;
    display: block;
    padding: .938rem 0;
    width: 100%;
    border-bottom: 1px solid $color--white;
    &:hover {
      color: $color--light;
      border-bottom: 1px solid $color--light;
    }
  }
}
@media only screen and(min-width:48em) {
  .search-wrapper {
    top: 100px;
  }
  .search-list .entry-date {
    display: inline;
  }
}
body.search-overlay {
  overflow: hidden;
  &:after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 9001;
    width: 100%;
    height: 100%;
    min-height: 100%;
    background-color: $color--black--overlay;
    content: '';
  }
}
