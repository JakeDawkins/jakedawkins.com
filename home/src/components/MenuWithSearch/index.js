import React, { Component, PropTypes } from "react";
import "./menuWithSearch.scss"
import { Link } from "react-router";

// XXX Can I move all aside styles here?

class MenuWithSearch extends Component {

  // Links unused currently
  static propTypes = {
    slide: PropTypes.bool,
    links: PropTypes.array,
  }

  constructor(props) {
    super(props);
    this.state = {
      slide: props.slide || false
    };

    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu(){
    console.log("toggle to ", !this.state.slide);
    this.setState({
      slide: !this.state.slide
    });
    //#sidebar add class slide
  }

  render(){
    return(
      <div>
        <div className="search-wrapper">
          <div className="search-form">
            <input type="text" className="search-field" placeholder="Search..." />
            <svg className="icon-remove-sign"><use xlinkHref="#icon-close"></use></svg>
            <ul className="search-results search-list"></ul>
          </div>
        </div>
        <div id="fade" className={this.state.slide ? "overlay slide" : "overlay"} onClick={this.toggleMenu}></div>
        <a id="slide" className={this.state.slide ? "slideButton fade slide" : "slideButton fade"}>
          <svg
            id="open"
            className="icon-menu"
            onClick={this.toggleMenu}
            style={this.state.slide ? {display:"none"} : {display: "inline"}}
          ><use xlinkHref="#icon-menu"></use></svg>
          <svg
            id="close"
            className="icon-menu"
            onClick={this.toggleMenu}
            style={this.state.slide ? {display:"inline"} : {display: "none"}}
          ><use xlinkHref="#icon-close"></use></svg>
        </a>
        <aside id="sidebar" className={this.state.slide ? "slide" : ""}>
        <nav id="navigation">
          <h2>MENU</h2>
          <ul>
            <li>
              <Link to="/posts">Posts</Link>
            </li>
            <li>
              <Link to="/about">About</Link>
            </li>
          </ul>
        </nav>
        </aside>
        {/*
        <a id="search" className="dosearch">
          <svg className="icon-menu icon-search"><use xlinkHref="#icon-search"></use></svg>
        </a>
        */}
      </div>
    );
  }
}

export default MenuWithSearch;
