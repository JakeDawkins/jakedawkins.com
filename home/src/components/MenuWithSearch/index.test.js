import MenuWithSearch from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("MenuWithSearch", () => {

  it("should exist", () => {
    const menuWithSearch = renderer.create(
      <MenuWithSearch />
    );

    expect(menuWithSearch).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    const menuWithSearch = renderer.create(
      <MenuWithSearch />
    );

    const menuWithSearchSlide = renderer.create(
      <MenuWithSearch slide={true}/>
    );

    expect(menuWithSearchSlide).toMatchSnapshot();
    expect(menuWithSearchSlide).not.toBe(menuWithSearch);
  });

});
