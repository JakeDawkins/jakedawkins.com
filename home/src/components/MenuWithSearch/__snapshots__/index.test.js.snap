exports[`MenuWithSearch should exist 1`] = `
<div>
  <div
    className="search-wrapper">
    <div
      className="search-form">
      <input
        className="search-field"
        placeholder="Search..."
        type="text" />
      <svg
        className="icon-remove-sign">
        <use
          xlinkHref="#icon-close" />
      </svg>
      <ul
        className="search-results search-list" />
    </div>
  </div>
  <div
    className="overlay"
    id="fade"
    onClick={[Function bound toggleMenu]} />
  <a
    className="slideButton fade"
    id="slide">
    <svg
      className="icon-menu"
      id="open"
      onClick={[Function bound toggleMenu]}
      style={
        Object {
          "display": "inline"
        }
      }>
      <use
        xlinkHref="#icon-menu" />
    </svg>
    <svg
      className="icon-menu"
      id="close"
      onClick={[Function bound toggleMenu]}
      style={
        Object {
          "display": "none"
        }
      }>
      <use
        xlinkHref="#icon-close" />
    </svg>
  </a>
  <aside
    className=""
    id="sidebar">
    <nav
      id="navigation">
      <h2>
        MENU
      </h2>
      <ul>
        <li>
          <a
            onClick={[Function bound handleClick]}
            style={Object {}}>
            Posts
          </a>
        </li>
        <li>
          <a
            onClick={[Function bound handleClick]}
            style={Object {}}>
            About
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;

exports[`MenuWithSearch should render props correctly 1`] = `
<div>
  <div
    className="search-wrapper">
    <div
      className="search-form">
      <input
        className="search-field"
        placeholder="Search..."
        type="text" />
      <svg
        className="icon-remove-sign">
        <use
          xlinkHref="#icon-close" />
      </svg>
      <ul
        className="search-results search-list" />
    </div>
  </div>
  <div
    className="overlay slide"
    id="fade"
    onClick={[Function bound toggleMenu]} />
  <a
    className="slideButton fade slide"
    id="slide">
    <svg
      className="icon-menu"
      id="open"
      onClick={[Function bound toggleMenu]}
      style={
        Object {
          "display": "none"
        }
      }>
      <use
        xlinkHref="#icon-menu" />
    </svg>
    <svg
      className="icon-menu"
      id="close"
      onClick={[Function bound toggleMenu]}
      style={
        Object {
          "display": "inline"
        }
      }>
      <use
        xlinkHref="#icon-close" />
    </svg>
  </a>
  <aside
    className="slide"
    id="sidebar">
    <nav
      id="navigation">
      <h2>
        MENU
      </h2>
      <ul>
        <li>
          <a
            onClick={[Function bound handleClick]}
            style={Object {}}>
            Posts
          </a>
        </li>
        <li>
          <a
            onClick={[Function bound handleClick]}
            style={Object {}}>
            About
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
